### PR TITLE
Compile error on Linux 2.6.32-754.6.3.el6.x86_64 /gcc 7.4/NetOps.cpp

### PR DESCRIPTION
**Pull request by deryck1031**
Compile error on Linux 2.6.32-754.6.3.el6.x86_64 /gcc 7.4/NetOps.cpp (issue #1011)

added below in NetOps.h

#ifndef FOLLY_HAVE_SENDMMSG
#define FOLLY_HAVE_SENDMMSG 1
#endif

### CODE REVIEW

**Comment by yfeldblum** [comment]
Why is this a fix?

**Comment by deryck1031** [comment]
This is to support older versions of Linux (2.6 in this case) that does not have sendmmsg.

File: folly/net/NetOps.h [hunk 1]
```diff
@@ -77,8 +77,10 @@ struct mmsghdr {
   unsigned int msg_len;
 };
 #else
+#ifndef FOLLY_HAVE_SENDMMSG
```
Comment (yfeldblum) [inline folly/net/NetOps.h]: But why is *this* the fix?

V.s. updating the `#if` line just above.

Comment (deryck1031) [reply folly/net/NetOps.h]: struct mmsghdr is already defined in /usr/include/bits/socket.h.  If update #if statement, we will get 'redefinition of struct mmsghdr' compilation error.

Comment (yfeldblum) [reply folly/net/NetOps.h]: What did you try?

Comment (deryck1031) [reply folly/net/NetOps.h]: I tried below:

#if (!__linux__) || (defined(__ANDROID__) && (__ANDROID_API__ < 21)) || (defined(HAS_NO_SENDMMSG))

Comment (yfeldblum) [reply folly/net/NetOps.h]: We may need to move this to a config check in `folly/CMake/FollyConfigChecks.cmake`.

Comment (deryck1031) [reply folly/net/NetOps.h]: Make sense, I think we need to make below changes:

1. remove define in NetOps.h
-#else
-#define FOLLY_HAVE_SENDMMSG 1

2. add  FOLLY_HAVE_SENDMMSG  to  CMake/folly-config.h.cmake
+#cmakedefine FOLLY_HAVE_SENDMMSG 1

3. update CMake/FollyConfigChecks.cmake
+check_symbol_exists(sendmmsg sys/socket.h FOLLY_HAVE_SENDMMSG)

I can create a PR if above changes look good to you.

Comment (yfeldblum) [reply folly/net/NetOps.h]: Yes something like this. Can force-update this PR if desired.

Comment (deryck1031) [reply folly/net/NetOps.h]: there was something wrong with my old forked repo, so I have created a new PR (1053).
https://github.com/facebook/folly/pull/1053

**Pull request closed without merging**